src/persist-client/src/schema.rs

```diff
@@ -402,15 +404,98 @@ impl<K: Codec, V: Codec> PartMigration<K, V> {
     }
 }
 
+/// Returns if `new` is at least as nullable as `old`.
+///
+/// Errors if `new` is less nullable than `old`, or `old` and `new` are different types or have
+/// different nested fields.
+pub(crate) fn is_atleast_as_nullable(old: &DataType, new: &DataType) -> Result<(), anyhow::Error> {
```

Comment:
It is nearly identical! We could try and re-use `backward_compatible_typ(...)` but because `bacward_compatible_typ(...)` allows adding new columns is why I shied away from it.

I do believe we should be able to delete this code after a release or two since the new DataTypes will be migrated, so maybe code reuse isn't super important if we're going to get rid of it?

Reply:
Okay! I don't know that allowing additions is so much worse than allowing deletes, since neither should come up in practice... but if you're confident we can get rid of this in short order I won't worry about it too much one way or the other.